src/annotations/components/EditAnnotationOverlay.tsx

```diff
@@ -33,15 +33,15 @@ export const EditAnnotationOverlay: FC = () => {
   const dispatch = useDispatch()
   const {clickedAnnotation} = useSelector(getOverlayParams)
 
-  const handleSubmit = (editedAnnotation: EditAnnotation): void => {
+  const handleSubmit = async (editedAnnotation: EditAnnotation): Promise<void> => {
     try {
-      dispatch(editAnnotation(editedAnnotation))
+      await dispatch(editAnnotation(editedAnnotation))
       event('xyplot.annotations.edit_annotation.success')
       dispatch(notify(editAnnotationSuccess()))
       onClose()
     } catch (err) {
       event('xyplot.annotations.edit_annotation.failure')
-      dispatch(notify(editAnnotationFailed(err)))
+      dispatch(notify(editAnnotationFailed(err.response.data.message)))
```

Comment:
there's no guarentee that the error caught using this method was created by an invalid fetch on the requested api. could have been caused by a code branch error anywhere in the editAnnotation action, or even at an auth layer on the api. 

something like `err.response?.data?.message || err` should be good enough to cover those edge cases

Reply:
There should be a utility function that's called `getErrorMessage` or something that could be useful in getting the relevant information

Reply:
@Asalem1 can it live in the same `EditAnnotationOverlay.tsx` file?

Reply:
High level, I don't see why not. It seems like this function should be a thunk that exists in a `thunks` file, but that's besides the point.

Reply:
For sure, I'll add a function in the EditAnnotationOverlay.tsx file for now.